src/features/in-app-messenger/contextual/templates/PreviewTitleBodyTwoButtonsModalTemplate.js

```diff
@@ -0,0 +1,98 @@
+// @flow
+
+/**
+ * This component is intended to be passed to TemplateContainer as templateComponent prop, like this:
+ *   <TemplateContainer templateID='my-template' templateComponent=MyTemplate/>
+ */
+
+import * as React from 'react';
+
+import PrimaryButton from '../../../../components/primary-button';
+import { Modal, ModalActions } from '../../../../components/modal';
+import Button from '../../../../components/button';
+import MessagePreviewContent from '../../../message-preview-content/MessagePreviewContent';
+import { messageActions } from '../../types/message-actions';
+import { MessageActions, type PreviewTitleBodyTwoButtonsModalParams } from '../../types';
+
+import './styles/PreviewTitleBodyTwoButtonsModalTemplate.scss';
+
+type Props = {
+    apiHost: string,
+    contentPreviewProps?: ContentPreviewProps,
+    getToken: (folderID: string | number) => Promise<Token>,
+    onAction: (MessageActions, ...Array<any>) => any,
+    params: PreviewTitleBodyTwoButtonsModalParams,
+};
+
+const handleClose = onAction => {
+    onAction([messageActions.close]);
+};
+
+const handleButton1Click = (onAction, button1) => {
+    if (button1) {
+        onAction(button1.actions);
+    }
+};
+
+const handleButton2Click = (onAction, button2) => {
+    if (button2) {
+        onAction(button2.actions);
+    }
+};
+
+const PreviewTitleBodyTwoButtonsModalTemplate = ({
+    apiHost,
+    contentPreviewProps,
```

Comment:
do we need a default prop since it's optional?

Reply:
I don't think so, we are passing these props to `ContentPreview`, in that way we can provide any props needed for it, but I believe nothing is really required